pkg/ddevapp/codeigniter.go

```diff
@@ -0,0 +1,167 @@
+package ddevapp
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/ddev/ddev/pkg/archive"
+	"github.com/ddev/ddev/pkg/fileutil"
+	"github.com/ddev/ddev/pkg/nodeps"
+)
+
+// createCodeIgniterSettingsFile creates/updates the .env file for CodeIgniter 4
+func createCodeIgniterSettingsFile(app *DdevApp) (string, error) {
+	envFilePath := filepath.Join(app.AppRoot, ".env")
```

Comment:
Please update all instances of using `filepath.Join(app.AppRoot, ...)` with `filepath.Join(app.AppRoot, app.ComposerRoot, ...)`

```suggestion
	envFilePath := filepath.Join(app.AppRoot, app.ComposerRoot, ".env")
```

This is necessary to properly manage `.env` when the project is created in a subfolder.

More details on this in:

- #7558